src/main/kotlin/com/learnspigot/bot/profile/ProfileCommand.kt

```diff
@@ -21,21 +24,58 @@ class ProfileCommand {
     )
     fun onProfileCommand(
         event: SlashCommandInteractionEvent,
-        @Description("User to show profile") @Optional user: User?
+        @Description("User to show profile") @Optional user: User?,
+        @Description("URL to show profile") @Optional url: String?
     ) {
-        val finalUser = user ?: event.user
-        val profile = profileRegistry.findByUser(finalUser)
-
-        event.replyEmbeds(
-            embed()
-                .setTitle("Profile Lookup")
-                .addField("Discord", finalUser.name + " (" + finalUser.asMention + ")", false)
-                .addField("Udemy", profile.udemyProfileUrl ?: "Not linked", false)
-                .addField("Reputation", profile.reputation.size.toString(), true)
-                .addField("(Notifications)", profile.notifyOnRep.toString(), true)
-                .setThumbnail(finalUser.effectiveAvatarUrl)
-                .build()
-        ).setEphemeral(true).queue()
+        val profileByURL = profileRegistry.findByURL(url ?: "")
+
+        val embed: MessageEmbed = when {
```

Comment:
I like the change to a when statement, but this can be improved even further with more concise conditions, I'll leave separate examples further on